chichiwang/exercise-js
# Track `project-configs.js` in git

## Description
Linked to Issue: #125

Remove `project-configs.js` from `.gitignore` and replace `project-configs-example.js` with it. The reason being that this config is being re-positioned to not hold local secrets, but project-based environment-specific configurations that should be tracked for both local development and production deployment.

With the removal of `project-config-example.js`, the npm script `create-config` is no longer necessary. Documentation is updated to reflect this change. Documentation of baseline, boilerplate, project configuration variables is added as well.

## Changes
* Remove `project-configs.js` from `.gitignore`
* Update `docs/README.md`
  * Remove step to run `create-config` from section: Starting a Project
  * Remove documentation for `create-config` from section: NPM Scripts
  * Update section "Project Configurations" to
    * Remove references to `project-configs-example.js`
    * Document baseline project config variables
    * Document using a `/* global */` comment for project config variables
* Remove script `create-config` from `package.json`
* Remove documentation for script `create-config` from `pages/Documentation/NPMScripts.tsx`
* Update `pages/Documentation/ProjectConfigurations.tsx` to reflect the changes to `docs/README.md`
* Remove step to run `npm run create-config` in `pages/Documentation/StartingAProject.tsx`
* Update `.list li` `line-height` to `1.75rem` in `pages/Documentation/styles.module.scss`
* Replace `project-configs-example.js` with `project-configs.js`
  * Set `SERVE_STATIC` to `false` for production builds

## Steps to QA
* View and verify that the changes in `docs/README.md` look good and make sense
* Pull down and switch to this branch
* Verify the project configs work for the Express server
* Verify the changes to the `/documentation` page look good and make sense in the browser
  * Ensure the new code snippet in the section "Project Configurations" looks correct, and the copy button copies the correct code
